presto-client/src/main/java/com/facebook/presto/client/QueryActions.java

```diff
@@ -0,0 +1,164 @@
+            @JsonProperty("setSessionProperties") Map<String, String> setSessionProperties,
+            @JsonProperty("clearSessionProperties") Set<String> clearSessionProperties,
+            @JsonProperty("addedPreparedStatements") Map<String, String> addedPreparedStatements,
+            @JsonProperty("deallocatedPreparedStatements") Set<String> deallocatedPreparedStatements,
+            @JsonProperty("startedTransactionId") String startedTransactionId,
+            @JsonProperty("clearTransactionId") Boolean clearTransactionId)
+    {
+        this.setCatalog = setCatalog;
+        this.setSchema = setSchema;
+        this.setSessionProperties = setSessionProperties != null ? ImmutableMap.copyOf(setSessionProperties) : null;
+        this.clearSessionProperties = clearSessionProperties != null ? ImmutableSet.copyOf(clearSessionProperties) : null;
+        this.addedPreparedStatements = addedPreparedStatements != null ? ImmutableMap.copyOf(addedPreparedStatements) : null;
+        this.deallocatedPreparedStatements = deallocatedPreparedStatements != null ? ImmutableSet.copyOf(deallocatedPreparedStatements) : null;
+        this.startedTransactionId = startedTransactionId;
+        this.clearTransactionId = clearTransactionId;
+    }
+
+    @Nullable
+    @JsonProperty
+    public String getSetCatalog()
+    {
+        return setCatalog;
+    }
+
+    @Nullable
+    @JsonProperty
+    public String getSetSchema()
+    {
+        return setSchema;
+    }
+
+    @Nullable
+    @JsonProperty
+    public Map<String, String> getSetSessionProperties()
+    {
+        return setSessionProperties;
+    }
+
+    @Nullable
+    @JsonProperty
+    public Set<String> getClearSessionProperties()
+    {
+        return clearSessionProperties;
+    }
+
+    @Nullable
+    @JsonProperty
+    public Map<String, String> getAddedPreparedStatements()
+    {
+        return addedPreparedStatements;
+    }
+
+    @Nullable
+    @JsonProperty
+    public Set<String> getDeallocatedPreparedStatements()
+    {
+        return deallocatedPreparedStatements;
+    }
+
+    @Nullable
+    @JsonProperty
+    public String getStartedTransactionId()
+    {
+        return startedTransactionId;
+    }
+
+    @Nullable
+    @JsonProperty
+    public Boolean isClearTransactionId()
+    {
+        return clearTransactionId;
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("setSessionProperties", setSessionProperties)
+                .add("clearSessionProperties", clearSessionProperties)
+                .add("addedPreparedStatements", addedPreparedStatements)
+                .add("deallocatedPreparedStatements", deallocatedPreparedStatements)
+                .add("startedTransactionId", startedTransactionId)
+                .add("clearTransactionId", clearTransactionId)
+                .toString();
+    }
+
+    @Nullable
+    public static QueryActions createIfNecessary(
```

Comment:
So, that it can directly be passed as a field value in `QueryResults`. Also, it seems to be more inline with the current style of using nullable fields for classes used as json structures.